From 681f64146a3a0f1cc717c70311fd322da4c7a74d Mon Sep 17 00:00:00 2001
From: Wan Ahmad Zainie <wan.ahmad.zainie.wan.mohamad@intel.com>
Date: Fri, 24 May 2019 10:46:52 +0800
Subject: [PATCH 44/44] cpufreq: intel_tfm_gov: convert to use kobj_attribute

This patch will fix the error cause by backporting upstream commit
625c85a62cb7 ("cpufreq: Use struct kobj_attribute instead of struct
global_attr").

Signed-off-by: Wan Ahmad Zainie <wan.ahmad.zainie.wan.mohamad@intel.com>
---
 drivers/cpufreq/intel_tfm_governor.c | 24 ++++++++++++------------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/drivers/cpufreq/intel_tfm_governor.c b/drivers/cpufreq/intel_tfm_governor.c
index ae810d8..cc67771 100644
--- a/drivers/cpufreq/intel_tfm_governor.c
+++ b/drivers/cpufreq/intel_tfm_governor.c
@@ -142,7 +142,7 @@ void gpu_unregister_notification(void)
 }
 
 static ssize_t show_gpu_freq_stats(struct kobject *kobj,
-		struct attribute *attr, char *buf)
+		struct kobj_attribute *attr, char *buf)
 {
 	ssize_t ret = 0;
 	int i;
@@ -167,7 +167,7 @@ static ssize_t show_gpu_freq_stats(struct kobject *kobj,
 }
 
 static ssize_t show_tfm_gpu_budget_ms(struct kobject *kobj,
-		struct attribute *attr, char *buf)
+		struct kobj_attribute *attr, char *buf)
 {
 	ssize_t ret = 0;
 
@@ -178,7 +178,7 @@ static ssize_t show_tfm_gpu_budget_ms(struct kobject *kobj,
 }
 
 static ssize_t store_tfm_gpu_budget_ms(struct kobject *kobj,
-		struct attribute *attr, const char *buf, size_t count)
+		struct kobj_attribute *attr, const char *buf, size_t count)
 {
 	int ret;
 	long input;
@@ -195,7 +195,7 @@ static ssize_t store_tfm_gpu_budget_ms(struct kobject *kobj,
 }
 
 static ssize_t show_gpu_tfm_time_ms(struct kobject *kobj,
-		struct attribute *attr, char *buf)
+		struct kobj_attribute *attr, char *buf)
 {
 	ssize_t ret = 0;
 
@@ -509,7 +509,7 @@ void cpu_unregister_notification(void)
 }
 
 static ssize_t show_cpu_freq_stats(struct kobject *kobj,
-		struct attribute *attr, char *buf)
+		struct kobj_attribute *attr, char *buf)
 {
 	ssize_t ret = 0;
 	int i, j;
@@ -569,7 +569,7 @@ static ssize_t show_cpu_freq_stats(struct kobject *kobj,
 }
 
 static ssize_t show_cpu_tfm_time_ms(struct kobject *kobj,
-		struct attribute *attr, char *buf)
+		struct kobj_attribute *attr, char *buf)
 {
 	ssize_t ret;
 
@@ -579,7 +579,7 @@ static ssize_t show_cpu_tfm_time_ms(struct kobject *kobj,
 }
 
 static ssize_t show_tfm_cpu_budget_ms(struct kobject *kobj,
-		struct attribute *attr, char *buf)
+		struct kobj_attribute *attr, char *buf)
 {
 	ssize_t ret = 0;
 
@@ -590,7 +590,7 @@ static ssize_t show_tfm_cpu_budget_ms(struct kobject *kobj,
 }
 
 static ssize_t store_tfm_cpu_budget_ms(struct kobject *kobj,
-		struct attribute *attr, const char *buf, size_t count)
+		struct kobj_attribute *attr, const char *buf, size_t count)
 {
 	int ret;
 	long input;
@@ -958,7 +958,7 @@ static void tfmg_sysfs_cleanup(void)
 }
 
 static ssize_t show_tfm_control(struct kobject *kobj,
-		struct attribute *attr, char *buf)
+		struct kobj_attribute *attr, char *buf)
 {
 	ssize_t ret;
 
@@ -968,7 +968,7 @@ static ssize_t show_tfm_control(struct kobject *kobj,
 }
 
 static ssize_t store_tfm_control(struct kobject *kobj,
-		struct attribute *attr, const char *buf, size_t count)
+		struct kobj_attribute *attr, const char *buf, size_t count)
 {
 	int ret;
 	long input;
@@ -991,7 +991,7 @@ static ssize_t store_tfm_control(struct kobject *kobj,
 }
 
 static ssize_t show_tfm_hysteresis_ms(struct kobject *kobj,
-		struct attribute *attr, char *buf)
+		struct kobj_attribute *attr, char *buf)
 {
 	ssize_t ret;
 
@@ -1002,7 +1002,7 @@ static ssize_t show_tfm_hysteresis_ms(struct kobject *kobj,
 }
 
 static ssize_t store_tfm_hysteresis_ms(struct kobject *kobj,
-		struct attribute *attr, const char *buf, size_t count)
+		struct kobj_attribute *attr, const char *buf, size_t count)
 {
 	int ret;
 	long input;
-- 
1.9.1

